### ISSUE TITLE
Mardown in JSDoc from TypeScript .d.ts lib treated as plain text

### ISSUE THREAD

**Issue by mjbvz** [microsoft/monaco-editor#1079]
_From @trevorade on September 28, 2018 21:30_

- VSCode Version: 1.27.1
- Monaco Version: 0.14.3

Steps to Reproduce:

1.  Register a .d.ts file with `monaco.languages.typescript.FOOscriptDefaults.addExtraLib`
2.  Trigger documentation in autocomplete or hover.  Markdown in JsDocs of the .d.ts file is not rendered:
    ![image](https://user-images.githubusercontent.com/13050011/46234216-d38fd500-c332-11e8-8029-1ecc3409a4a4.png)

The unwanted asterisks in the MarkDown code block is tracked by Microsoft/TypeScript#23517 but the larger issue here is separate.

Here's the code I'm using in the [Monaco Playground](https://microsoft.github.io/monaco-editor/playground.html):

```javascript
monaco.languages.typescript.javascriptDefaults.addExtraLib(
    `/**
 * Function documentation **bold** _italics_.
 * 
 * Example :
 * 
 * \`\`\`
 * aaExampleFunction(0);
 * \`\`\`
 * 
 * More markdown text
 * 
 * @param num Parameter description
 * @return Return type description.
 */
function aaExampleFunction(num: number): number;`,
    'myDefault:some.file.d.ts'
);

monaco.languages.typescript.javascriptDefaults.setCompilerOptions({
    allowJs: true,
    allowNonTsExtensions: true,
    target: monaco.languages.typescript.ScriptTarget.ESNext,
    checkJs: true,
    lib: ['myDefault:some.file.d.ts'],
});

monaco.editor.create(document.getElementById("container"), {
    value: "",
    language: "javascript"
});
```

_Copied from original issue: Microsoft/vscode#59654_

**Comment by spahnke** [comment]
Seems to be related to #1074

**Comment by spahnke** [comment]
> The function declaration on the first line has no syntax highlighting as compared to the left

I wanted to address that in my PR (Microsoft/monaco-typescript#22) as well but the `detail` field of a
`monaco.languages.CompletionItem` only takes a `string` and no `monaco.IMarkdownString`. So AFAIKT this is not possible at the moment. (VS Code currently doesn't highlight that part either)

**Comment by trevorade** [comment]
Here's the feature request for Markdown in CompletionItem details:
https://github.com/Microsoft/language-server-protocol/issues/437

Looks like you can fake this somewhat by just using the `documentation` and leaving the description blank.  The `---` horizontal rule here doesn't look as nice though:

![image](https://user-images.githubusercontent.com/13050011/46302859-85afe280-c567-11e8-8e85-3031a0f85cb8.png)

Relevant [playground](https://microsoft.github.io/monaco-editor/playground.html) code:

```js
monaco.languages.registerCompletionItemProvider('javascript', {
    provideCompletionItems: function(model, position) {
        return [{
            label: 'aaaExampleFunction',
            kind: monaco.languages.CompletionItemKind.Function,
            documentation: {
                value:
`\`\`\`typescript
function aaExampleFunction(num: number): number
\`\`\`

---

Function documentation **bold** _italics_.
 
Example :
 
\`\`\`js
aaExampleFunction(0);
\`\`\`
 
More markdown text

**@param num** Parameter description  
**@return Return** type description.
`
            },
            insertText: '"lodash": "*"'
        }];
    }
});

monaco.editor.create(document.getElementById("container"), {
    value: "",
    language: "javascript"
});
```

**Comment by mjbvz** [comment]
I just merged a PR that fixes #1074. Can you please please confirm if it address this issue as well

**Comment by trevorade** [comment]
> I just merged a PR that fixes #1074. Can you please please confirm if it address this issue as well

@mjbvz, I don't have a development environment setup for VsCode/Monaco so I can't verify this until it shows up in a VsCode or Monaco release.

I'll post back later though.

**Comment by alexdima** [comment]
This has been fixed in the meantime.

![image](https://user-images.githubusercontent.com/5047891/70711861-bf8b8580-1ce2-11ea-9a8d-45ce28e9e669.png)
